File: src/main/java/spixie/RootTimeChanger.java
package spixie;

public class RootTimeChanger implements ValueChanger {
    private Value frame;
    private Value bpm;
    private Value time;

    public RootTimeChanger(Value frame, Value bpm, Value time) {
        this.frame = frame;
        this.bpm = bpm;
        this.time = time;
    }

    @Override
    public void updateOutValue() {
        time.set(bpm.getFraction().divide(3600).multiply(frame.getFraction()));
    }
}

File: src/main/kotlin/spixie/RootTimeChanger.kt
package spixie

class RootTimeChanger(private val frame: Value, private val bpm: Value, private val time: Value) : ValueChanger {

    override fun updateOutValue() {
        time.set(bpm.fraction.divide(3600).multiply(frame.fraction))
    }

    override val valueToBeChanged: Value.Item
        get() = Value.EMPTY.item()
}
